app/services/registration_checker.rb

```diff
@@ -173,7 +173,8 @@ def validate_update_status!
 
       raise RegistrationError.new(:unprocessable_entity, ErrorCodes::INVALID_REQUEST_DATA) unless Registration::REGISTRATION_STATES.include?(new_status)
       raise RegistrationError.new(:forbidden, ErrorCodes::COMPETITOR_LIMIT_REACHED) if
-        new_status == 'accepted' && Registration.accepted_competitors_count(@competition_info.competition_id) == @competition_info.competitor_limit
+        new_status == 'accepted' && Registration.accepted_competitors_count(@competition_info.competition_id) == @competition_info.competitor_limit &&
+        @competition_info.competitor_limit != 0
```

Comment:
1. `competitor_limit > 0`
2. You can put this check first, because the (relatively) expensive `Registration.accepted_competitors_count` doesn't even need to be run if there is no competitor limit